### PR TITLE
:speedboat: intiial work on search while typing

### PR DESCRIPTION
**Pull request by markysoft**
Adding AJAX functionality with fallback to no script version.

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-89 in nhsuk-dev succeeded (http://gp-finder-pr-89.dev.beta.nhschoices.net)

File: app/views/results.nunjucks [hunk 1]
```diff
@@ -26,20 +26,7 @@
 
     <a href="{{ SITE_ROOT }}" class="link-back">Back</a>
     <h1 class="sr-only" data-results="{{gps.length}}">Search results</h1>
-    <h2 class="results__header">GP {% if gps.length === 1 %}surgery{% else %}surgeries{%endif %} matching '{{ search }}' </h2>
```
Comment (beseven) [inline app/views/results.nunjucks]: Question for @mcgrathjoe / interaction designers / looking at the prototype: 
- what do we do with no results via the JS solution
- what do we do with an empty search string via the JS solution
- how do we test the js solution

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-89 in nhsuk-dev succeeded (http://gp-finder-pr-89.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-89 in nhsuk-dev succeeded (http://gp-finder-pr-89.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-89 in nhsuk-dev succeeded (http://gp-finder-pr-89.dev.beta.nhschoices.net)

**Comment by beseven** [comment]
Superseded by https://github.com/nhsuk/gp-finder/pull/91

**Pull request closed without merging**